src/Microsoft.Azure.WebJobs/FunctionNameAttribute.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.Azure.WebJobs
+{
+    /// <summary>
+    /// This is for providing a custom name to for an SDK function.
```

Comment:
All features we add to the core SDK are usable in the SDK outside of functions. So based on some of the changes above, it looks like you're overriding function indexing behavior, etc. We use function name in lots of other places too, including in the Dashboard, etc. So I'm just saying that if we're adding a new core SDK feature we need to verify it end to end in all the various usages.